jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessors.java

```diff
@@ -65,4 +80,40 @@ private static String determineJarType(Path jarPath) throws IOException {
       return STANDARD;
     }
   }
+
+  /**
+   * Determines the java version of JAR. Derives the version from the first .class file it finds in
+   * the JAR.
+   *
+   * @param jarPath path to the jar
+   * @return java version
+   * @throws IOException if I/O exception thrown when opening the jar file
+   */
+  private static Integer getJavaMajorVersion(Path jarPath) throws IOException {
+    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
+      Enumeration<JarEntry> jarEntries = jarFile.entries();
+      while (jarEntries.hasMoreElements()) {
+        String jarEntry = jarEntries.nextElement().toString();
+        if (jarEntry.endsWith(".class") && !jarEntry.equals("module-info.class")) {
```

Comment:
Is `jarEntry` a filename without path? If not, maybe it should be `!jarEntry.endsWith("module-info.class")`.

Reply:
good catch, thanks!